service/src/test/java/uk/gov/hmcts/reform/fpl/validators/HasDocumentStatusValidatorTest.java

```diff
@@ -0,0 +1,115 @@
+package uk.gov.hmcts.reform.fpl.validators;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+
+import javax.validation.ConstraintValidatorContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(SpringExtension.class)
+class HasDocumentStatusValidatorTest {
+    HasDocumentStatusValidator validator = new HasDocumentStatusValidator();
+
+    @Mock
+    private ConstraintValidatorContext constraintValidatorContext;
+
+    @Mock
+    private ConstraintValidatorContext.ConstraintViolationBuilder constraintViolationBuilder;
+
+    @Mock
+    private ConstraintValidatorContext.ConstraintViolationBuilder.NodeBuilderCustomizableContext
+        nodeBuilderCustomizableContext;
+
```

Comment:
why all the mocks?

Reply:
If you look at the validatior, EPOReason validator for example I am setting the property path so I can find it for section building in the service. Because of this I had to call a few methods on ConstraintValidatorContext which needed mocked

Reply:
I remember we discussed this my recommendation would be to use the real validator things targeting your class. This still gives you segregation, i.e.:

```
private Validator validator;

    @BeforeEach
    public void setup() {
        validator = Validation.buildDefaultValidatorFactory()
            .getValidator();
    }

        Children children = Children.builder().build();

        Set<ConstraintViolation<Children>> validationIssues = validator.validate(children);

        assertThat(validationIssues()).isFalse();
//other assertions, like checking the property

```
The above is nicer and allows you to test the properties as well. I think you should try it.